pymeasure/instruments/anritsu/anritsuMS2090A.py

```diff
@@ -0,0 +1,513 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    truncated_range
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class AnritsuMS2090A(Instrument):
+    """Anritsu MS2090A Handheld Spectrum Analyzer."""
+
+    def __init__(self, adapter, name="Anritsu MS2090A Handheld Spectrum Analyzer", **kwargs):
+        self.analysis_mode = None
+        super().__init__(
+            adapter, name, **kwargs)
+
+    #############
+    #  Mappings #
+    #############
+
+    ONOFF = {True: 'ON', False: 'OFF'}
+    OFFFIRSTREPEAT = ['OFF', 'FIRSt', 'REPeat']
+    SPAMODES = ["SPECtrum", "NRADio", "RTSA", "LTE", "EMFMeter", "PANalyzer"]
+
+    ####################################
+    #              GPS                 #
+    ####################################
+
+    gps_full = Instrument.measurement(
+        "FETCh:GPS:FULL?",
+        """
+        Returns the timestamp, latitude, longitude, altitude, and satellite count of the device.
+        """
+    )
+
+    gps_all = Instrument.measurement(
+        "FETCh:GPS:ALL?",
+        """Returns the fix timestamp, latitude, longitude, altitude and information on the sat used.
+        """
+
+    )
+
+    gps = Instrument.measurement(
+        ":FETCh:GPS?",
+        """
+        Returns the timestamp, latitude, and longitude of the device.
+        """
+    )
+
+    gps_last = Instrument.measurement(
+        ":FETCh:GPS:LAST?",
+        """
+        Returns the timestamp, latitude, longitude, and altitude of the last fixed GPS result.
+        """
+    )
+
+    external_current = Instrument.measurement(
+        "BIAS:EXT:CURR?",
+        """
+        This command queries the actual bias current in A
+        """
+    )
+
+    ####################################
+    # Spectrum Parameters - Wavelength #
+    ####################################
+
+    frequency_center = Instrument.control(
+        "FREQuency:CENTer?", "FREQuency:CENTer %g",
+        "Sets the center frequency in Hz",
+        validator=truncated_range,
+        values=[-99999999995, 299999999995]

```

Comment:
Would it not technically make more sense to limit this range to [9000, 54E9] per the specifiaction of the instrument? 
Also it might be better to use the `strict_range` type of validator